src/DynamoRevit/Models/RevitDynamoModel.cs

```diff
@@ -389,18 +390,31 @@ private void SetupPython()
                 (a, b, c, d, e) => ElementBinder.IsEnabled = false;
             IronPythonEvaluator.EvaluationEnd += (a, b, c, d, e) => ElementBinder.IsEnabled = true;
 
+            var marshaler = new DataMarshaler();
+            marshaler.RegisterMarshaler(
+                (Revit.Elements.Element element) => element.InternalElement);
+            marshaler.RegisterMarshaler((Category element) => element.InternalCategory);
+            Func<object, object> unwrap = marshaler.Marshal;
             // register UnwrapElement method in ironpython
             IronPythonEvaluator.EvaluationBegin += (a, b, scope, d, e) =>
             {
-                var marshaler = new DataMarshaler();
-                marshaler.RegisterMarshaler(
-                    (Revit.Elements.Element element) => element.InternalElement);
-                marshaler.RegisterMarshaler((Category element) => element.InternalCategory);
-
-                Func<object, object> unwrap = marshaler.Marshal;
                 scope.SetVariable("UnwrapElement", unwrap);
             };
 
+            CPythonEvaluator.OutputMarshaler.RegisterMarshaler(
+                (Element element) => element.ToDSType(true));
+
+            // Turn off element binding during cpython script execution
+            CPythonEvaluator.EvaluationBegin +=
+                (a, b, c, d) => ElementBinder.IsEnabled = false;
+            CPythonEvaluator.EvaluationEnd += (a, b, c, d) => ElementBinder.IsEnabled = true;
+
+            // register UnwrapElement method in cpython
+            CPythonEvaluator.EvaluationBegin += (a, scope, c, d) =>
```

Comment:
should we be unsubscribing these when python nodes are removed from the workspace?

Reply:
I didn't get that @mjkkirschner . These events are static, not really associated to a Python node in particular.

Reply:
whoops! my mistake.